narwhal/node/src/lib.rs

```diff
@@ -111,6 +112,14 @@ impl Node {
             .await?;
 
             handles.extend(consensus_handles);
+            last_committed_round = store
```

Comment:
Nothing wrong with this approach, but to avoid introducing one more parameter now in the `primary::spawn` and another branch for update, we could re-use the existing mechanisms in `Consensus` and `StateHandler` for this. Meaning, Consensus could emit after recovery the last committed round, and `StateHandler` via a recovery method - similar to Core - could consume this change and update the `tx_consensus_round_updates` in a blocking fashion to ensure that all the downstream components will use.

Reply:
I considered that approach, but it sees reading the last committed the rounds from store, making sure there is 1 and only 1 origin at the highest round, re-send this certificate assuming it is the last sent certificate etc seems a bit brittle. There are a number of other listeners to `tx_committed_certificates` which I did not have time to verify that resending is ok.

I think the short-to-medium term strategy can be to delete StateHandler and have the Consensus component publish via `tx_consensus_round_updates` itself. I can refactor in this PR. Longer term I believe there are plans to have bigger refactors across narwhal and consensus.